voltixApp/VoltixApp/View Models/SendTransaction.swift

```diff
@@ -37,21 +37,75 @@ class SendTransaction: ObservableObject, Hashable {
     var amountInWei: BigInt {
         BigInt(amountDecimal * pow(10, Double(EVMHelper.ethDecimals)))
     }
+    
+    var isAmountExceeded: Bool {
+        
+        let totalBalance = BigInt(coin.rawBalance) ?? BigInt.zero
+        
+        var gasInt = BigInt.zero
+        if coin.isNativeToken {
+            gasInt = BigInt(gas) ?? BigInt.zero
+            if coin.chainType == .EVM {
+                if let gasLimitBigInt = BigInt(coin.feeDefault) {
+                    gasInt = gasInt * gasLimitBigInt
+                }
+            }
+        }
+        
+        let totalTransactionCost = amountInRaw + gasInt
+        
+        return totalTransactionCost > totalBalance
+        
+    }
+    
+    var hasEnoughNativeTokensToPayTheFees: Bool {
+        var gasInt = BigInt(gas) ?? BigInt.zero
+        
+        if coin.chainType == .EVM {
```

Comment:
This is not correct , many blockchains have multiple tokens , not only EVM.

For example solana has many tokens.  mayachain has both CACAO and Maya token

you should take the chain from the coin they are sending , and then find out whether they have any native token on that blockchain

Reply:
Yes, I can check a nonnative and get the native . It will work for all